File: DSA/Algorithm/DP/Subsequence.cs
// Online C# Editor for free
// Write, Edit and Run your C# code using C# Online Compiler

using System;

public class HelloWorld
{
    public static void Main(string[] args)
    {
        int[] arr = {1,2,3,4,5};
        SubSeq(arr);
        subSeqRecursive(0, new List<int>(), arr);
    }

  // Iterative using bitwise operator
    static void SubSeq(int[] arr){
     for(int i = 0; i<Math.Pow(2,arr.Length); i++){
         for(int j = 0; j<arr.Length; j++){
             if((i&(1<<j)) != 0){
                 Console.Write(arr[j]);
             }
         }
         Console.WriteLine();
     }   
    }

  static void subSeqRecursive(int index, List<int> res, int[] arr){
        if(index >= arr.Length){
            for(int i = 0; i<res.Count; i++)Console.Write(res[i] + " ");
            Console.WriteLine();
            return; 
        }
        res.Add(arr[index]);
        subSeqRecursive(index+1,res ,arr);
        res.RemoveAt(res.Count-1);
        subSeqRecursive(index+1,res ,arr);
    }
  
}

File: DSA/Algorithm/DP/SubSequence.java
// Online Java Compiler
// Use this editor to write, compile and run your Java code online
import  java.util.*;
class Main {
    public static void main(String[] args) {
        System.out.println("Try programiz.pro");
        Subsequence sb = new Subsequence();
        sb.getAllSubSeq(new int[]{2,3,4,5,9});
    }
    
}

class Subsequence{
    public void getAllSubSeq(int[] arr){
        List<Integer> ls = new ArrayList<>();
        getAllSubSeqRec(arr, 0, ls);
    }
    public void getAllSubSeqRec(int[] arr, int index, List<Integer> ls){
        if(index >= arr.length){
            System.out.println(ls);
            return;
        }
        ls.add(arr[index]);
        getAllSubSeqRec(arr, index+1, ls);
        ls.remove(ls.size()-1);
        getAllSubSeqRec(arr, index+1, ls);
    }
}
